### DISCUSSION TITLE
Any idea to install app with espresso into Huawei phone?

### DISCUSSION THREAD

**Discussion by impvd** [AdevintaSpain/Barista#400] [Q&A] [answered]
Hey folks, I'm new to Espresso, and I love Barista, it's awesome!

I tried to write some test cases with Espresso, and run it in Pixel 4, which works well.

However, when I tried to make it work in Huawei phone, I got something like:

<img src="https://user-images.githubusercontent.com/37371804/119591647-93e25980-be09-11eb-8b13-9ecc2079ff30.png" width="300">

<img src="https://user-images.githubusercontent.com/37371804/119591659-99d83a80-be09-11eb-9ede-89834d2f3f6e.png" width="300">

If I choose UIAutomator, it's easy to just click the **INSTALL** and **CONTINUE**, I'm not sure if there're any ways to work around?

**Comment by Sloy** [comment]
I'm sorry @hellotli, I don't think this is related to Barista. There must be something weird with the Huawei OS.

Just checking, are you sure you're importing Barista and/or Espresso dependencies using `androidTestImplementation` instead of `implementation`? Because Espresso should not be included in the application APK.

Other than this, I don't think we can help you 😞 

Good luck!

**Comment by impvd** [comment, marked as answer]
Hi @Sloy , awesome thanks :)

I believe I'd figure it out, just make sure "Monitor apps installed by ADB" is off.

![image](https://user-images.githubusercontent.com/37371804/120066799-c1016700-c0aa-11eb-9f9b-419471e912d6.png)

**Reply by Sloy** [reply]
Great!! 🎉 I'm glad you figured it out :)
